### PR TITLE
[bitnami/mariadb] Adds support for network policies

### PR DESCRIPTION
**Pull request by rafariossaa**
**Description of the change**

Adds support for network policies

**Benefits**

The user is now able to enable and customize network policies

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/mariadb/templates/networkpolicy-ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/component: primary
```
Comment (pablogalegoc) [inline bitnami/mariadb/templates/networkpolicy-ingress.yaml:16]: As this Network Policy only applies to the primary instance I would move it to `bitnami/mariadb/templates/primary`,  WDYT?

File: bitnami/mariadb/values.yaml [hunk 2]
```diff
@@ -1032,3 +1032,96 @@ metrics:
     ## ref: https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#prometheusspec
     ##
     additionalLabels: {}
+
+## @section NetworkPolicy parameters
+
+## Add networkpolcies
```
Comment (pablogalegoc) [inline bitnami/mariadb/values.yaml:1038]:
```suggestion
## Add networkpolicies
```

File: bitnami/mariadb/values.yaml [hunk 3]
```diff
@@ -1032,3 +1032,96 @@ metrics:
...
+    ##   label: monitoring
+    ##
+    podSelector: {}
+    ## e.g:
+    ## namespaceSelector:
+    ##   label: monitoring
+    ##
+    namespaceSelector: {}
+  ## @param networkPolicy.ingressRules.primaryAccessOnlyFrom.enabled Enable ingress rule that makes primary mariadb nodes only accessible from a particular origin.
+  ## @param networkPolicy.ingressRules.primaryAccessOnlyFrom.namespaceSelector [object] Namespace selector label that is allowed to acces the primary node. This label will be used to identified the allowed namespace(s).
```
Comment (pablogalegoc) [inline bitnami/mariadb/values.yaml:1061]:
```suggestion
  ## @param networkPolicy.ingressRules.primaryAccessOnlyFrom.namespaceSelector [object] Namespace selector label that is allowed to access the primary node. This label will be used to identified the allowed namespace(s).
```

File: bitnami/mariadb/templates/secondary/networkpolicy-ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/component: secondary
+      {{- include "common.labels.standard" . | nindent 6 }}
+  ingress:
```
Comment (pablogalegoc) [inline bitnami/mariadb/templates/secondary/networkpolicy-ingress.yaml:18]: Shouldn't we have a rule to allow traffic from primary pods by default?

File: bitnami/mariadb/templates/networkpolicy-egress.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  podSelector:
+    matchLabels:
+      {{- include "common.labels.standard" . | nindent 6 }}
```
Comment (pablogalegoc) [inline bitnami/mariadb/templates/networkpolicy-egress.yaml:16]: In Drupal we used
https://github.com/bitnami/charts/blob/ac6addfc22edfb912999e90d5a40e7198b26d3b0/bitnami/drupal/templates/networkpolicy-egress.yaml#L14-L16
Should we change it here or there?

**Review by pablogalegoc — COMMENTED** [review]
Thanks @rafariossaa! I've left some comments below

File: bitnami/mariadb/templates/networkpolicy-egress.yaml (hunk 5 above)
Comment (rafariossaa) [reply bitnami/mariadb/templates/networkpolicy-egress.yaml:16]: In drupal you have a `drupal` and a `mariadbp` pod that have different labels. Here in mariadb, the primary and secondary nodes have in common:
```
      app.kubernetes.io/name: mariadb
      helm.sh/chart: mariadb-9.8.0
      app.kubernetes.io/instance: mymariadb
      app.kubernetes.io/managed-by: Helm
```
That is also why the node type is included [here](https://github.com/bitnami/charts/blob/9a5d142aecf4682252488b3aa470fb89d0da39cc/bitnami/mariadb/templates/primary/statefulset.yaml#L7) or [here](https://github.com/bitnami/charts/pull/8079/files#diff-1aef2b0588aaa2c7cc6be8cf6417283b9dbb2eeb46a0326e29054c17143c992eR16)

File: bitnami/mariadb/templates/networkpolicy-ingress.yaml (hunk 1 above)
Comment (rafariossaa) [reply bitnami/mariadb/templates/networkpolicy-ingress.yaml]: I am moving the file to that directory.

File: bitnami/mariadb/templates/secondary/networkpolicy-ingress.yaml (hunk 4 above)
Comment (rafariossaa) [reply bitnami/mariadb/templates/secondary/networkpolicy-ingress.yaml:18]: Not needed. In mariadb, the secondary nodes pulls the changes from the primary.

File: bitnami/mariadb/templates/networkpolicy-egress.yaml (hunk 5 above)
Comment (pablogalegoc) [reply bitnami/mariadb/templates/networkpolicy-egress.yaml:16]: Ah got it, thanks!

**Review by pablogalegoc — APPROVED** [review]
LGTM!

**Pull request merged**